testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -24,612 +23,419 @@ import (
 )
 
 var (
-	errSszCast             = errors.New("ssz response is not a byte array")
-	errJsonCast            = errors.New("json response has wrong structure")
-	errEmptyPrysmData      = errors.New("prysm data is empty")
-	errEmptyLighthouseData = errors.New("lighthouse data is empty")
+	errSszCast   = errors.New("SSZ response is not a byte array")
+	errEmptyData = errors.New("data is empty")
 )
 
-type metadata struct {
-	start            primitives.Epoch
-	basepath         string
-	params           func(encoding string, currentEpoch primitives.Epoch) []string
-	requestObject    interface{}
-	prysmResps       map[string]interface{}
-	lighthouseResps  map[string]interface{}
-	customEvaluation func(interface{}, interface{}) error
+const (
+	msgWrongJson = "JSON response has wrong structure, expected %T, got %T"
+)
+
+type meta interface {
+	getBasePath() string
+	sszEnabled() bool
+	enableSsz()
+	getSszResp() []byte
+	setSszResp(resp []byte)
+	getStart() primitives.Epoch
+	setStart(start primitives.Epoch)
+	getReq() interface{}
+	setReq(req interface{})
+	getPResp() interface{}
+	getLResp() interface{}
```

Comment:
It's not immediately obvious that these are Prysm and Lighthouse responses, would it be worth adding a comment for anyone that may not be familiar with the code?